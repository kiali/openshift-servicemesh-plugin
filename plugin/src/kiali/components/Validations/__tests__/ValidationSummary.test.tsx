import * as React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { ValidationSummary } from '../ValidationSummary';

const testScenario = (summary: any) => {
  const wrapper = shallow(summary);
  expect(shallowToJson(wrapper)).toBeDefined();
  expect(shallowToJson(wrapper)).toMatchSnapshot();
};

describe('ValidationSummary renders', () => {
  // Istio config validations
  it('success icon when all istio components are valid', () => {
    testScenario(<ValidationSummary id={'1'} errors={0} warnings={0} objectCount={1} type="istio" />);
    testScenario(<ValidationSummary id={'1'} errors={0} warnings={0} objectCount={4} type="istio" />);
  });

  it('warning icon when all istio components are valid', () => {
    testScenario(<ValidationSummary id={'2'} errors={0} warnings={1} objectCount={1} type="istio" />);
    testScenario(<ValidationSummary id={'2'} errors={0} warnings={1} objectCount={3} type="istio" />);
  });

  it('error icon when all istio components are valid', () => {
    testScenario(<ValidationSummary id={'3'} errors={1} warnings={0} objectCount={1} type="istio" />);
    testScenario(<ValidationSummary id={'3'} errors={1} warnings={2} objectCount={3} type="istio" />);
  });

  it('N/A when all istio components are valid', () => {
    testScenario(<ValidationSummary id={'4'} errors={0} warnings={0} objectCount={0} type="istio" />);
  });

  // Service validations
  it('success icon when all services are valid', () => {
    testScenario(<ValidationSummary id={'1'} errors={0} warnings={0} type="service" />);
  });

  it('warning icon when all services are valid', () => {
    testScenario(<ValidationSummary id={'2'} errors={0} warnings={1} type="service" />);
    testScenario(<ValidationSummary id={'2'} errors={0} warnings={2} type="service" />);
  });

  it('error icon when all services are valid', () => {
    testScenario(<ValidationSummary id={'3'} errors={1} warnings={0} type="service" />);
    testScenario(<ValidationSummary id={'3'} errors={1} warnings={2} type="service" />);
  });
});
