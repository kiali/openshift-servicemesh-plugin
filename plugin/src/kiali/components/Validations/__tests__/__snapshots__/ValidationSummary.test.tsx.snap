// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidationSummary renders N/A when all components are valid 1`] = `
<Tooltip
  aria-label="Validations list"
  content={
    <Text
      className="kiali_f7610ot"
    >
      No Istio config objects found
    </Text>
  }
  enableFlip={true}
  position="auto"
>
  <div
    style={
      Object {
        "display": "inline-block",
        "marginLeft": "5px",
      }
    }
  >
    N/A
  </div>
</Tooltip>
`;

exports[`ValidationSummary renders error icon when all components are valid 1`] = `
<Tooltip
  aria-label="Validations list"
  content={
    <React.Fragment>
      <Text
        component="p"
        style={
          Object {
            "textAlign": "left",
            "textEmphasis": "strong",
          }
        }
      >
        Istio config objects analyzed: 
        1
      </Text>
      <div
        className="kiali_fljtil"
      >
        <div>
          1 error found
        </div>
      </div>
    </React.Fragment>
  }
  enableFlip={true}
  position="auto"
>
  <Validation
    severity="error"
  />
</Tooltip>
`;

exports[`ValidationSummary renders error icon when all components are valid 2`] = `
<Tooltip
  aria-label="Validations list"
  content={
    <React.Fragment>
      <Text
        component="p"
        style={
          Object {
            "textAlign": "left",
            "textEmphasis": "strong",
          }
        }
      >
        Istio config objects analyzed: 
        3
      </Text>
      <div
        className="kiali_fljtil"
      >
        <div>
          1 error found
        </div>
        <div>
          2 warnings found
        </div>
      </div>
    </React.Fragment>
  }
  enableFlip={true}
  position="auto"
>
  <Validation
    severity="error"
  />
</Tooltip>
`;

exports[`ValidationSummary renders success icon when all components are valid 1`] = `
<Tooltip
  aria-label="Validations list"
  content={
    <React.Fragment>
      <Text
        component="p"
        style={
          Object {
            "textAlign": "left",
            "textEmphasis": "strong",
          }
        }
      >
        Istio config objects analyzed: 
        1
      </Text>
      <div
        className="kiali_fljtil"
      >
        <div>
          No issues found
        </div>
      </div>
    </React.Fragment>
  }
  enableFlip={true}
  position="auto"
>
  <Validation
    severity="correct"
  />
</Tooltip>
`;

exports[`ValidationSummary renders success icon when all components are valid 2`] = `
<Tooltip
  aria-label="Validations list"
  content={
    <React.Fragment>
      <Text
        component="p"
        style={
          Object {
            "textAlign": "left",
            "textEmphasis": "strong",
          }
        }
      >
        Istio config objects analyzed: 
        4
      </Text>
      <div
        className="kiali_fljtil"
      >
        <div>
          No issues found
        </div>
      </div>
    </React.Fragment>
  }
  enableFlip={true}
  position="auto"
>
  <Validation
    severity="correct"
  />
</Tooltip>
`;

exports[`ValidationSummary renders warning icon when all components are valid 1`] = `
<Tooltip
  aria-label="Validations list"
  content={
    <React.Fragment>
      <Text
        component="p"
        style={
          Object {
            "textAlign": "left",
            "textEmphasis": "strong",
          }
        }
      >
        Istio config objects analyzed: 
        1
      </Text>
      <div
        className="kiali_fljtil"
      >
        <div>
          1 warning found
        </div>
      </div>
    </React.Fragment>
  }
  enableFlip={true}
  position="auto"
>
  <Validation
    severity="warning"
  />
</Tooltip>
`;

exports[`ValidationSummary renders warning icon when all components are valid 2`] = `
<Tooltip
  aria-label="Validations list"
  content={
    <React.Fragment>
      <Text
        component="p"
        style={
          Object {
            "textAlign": "left",
            "textEmphasis": "strong",
          }
        }
      >
        Istio config objects analyzed: 
        3
      </Text>
      <div
        className="kiali_fljtil"
      >
        <div>
          1 warning found
        </div>
      </div>
    </React.Fragment>
  }
  enableFlip={true}
  position="auto"
>
  <Validation
    severity="warning"
  />
</Tooltip>
`;
